app/Actions/Photo/Extensions/ImageEditing.php

```diff
@@ -0,0 +1,176 @@
+
+			if (Configs::get_value('medium_2x') === '1') {
+				$this->resizePhoto($photo, 'medium2x', $mediumMaxWidth * 2, $mediumMaxHeight * 2, $frame_tmp);
+			}
+		}
+
+		$smallMaxWidth = intval(Configs::get_value('small_max_width'));
+		$smallMaxHeight = intval(Configs::get_value('small_max_height'));
+		$this->resizePhoto($photo, 'small', $smallMaxWidth, $smallMaxHeight, $frame_tmp);
+
+		if (Configs::get_value('small_2x') === '1') {
+			$this->resizePhoto($photo, 'small2x', $smallMaxWidth * 2, $smallMaxHeight * 2, $frame_tmp);
+		}
+	}
+
+	/**
+	 * @param Photo $photo
+	 *
+	 * @return string Path of the jpg file
+	 */
+	public function createJpgFromRaw(Photo $photo): string
+	{
+		// we need imagick to do the job
+		if (!Configs::hasImagick()) {
+			Logs::notice(__METHOD__, __LINE__, 'Saving JPG of raw file to failed: Imagick not installed.');
+
+			return '';
+		}
+
+		$filename = $photo->url;
+		$url = Storage::path('raw/' . $filename);
+		$ext = pathinfo($filename)['extension'];
+
+		// test if Imagaick supports the filetype
+		// Query return file extensions as all upper case
+		if (!in_array(strtoupper($ext), \Imagick::queryformats())) {
+			Logs::notice(__METHOD__, __LINE__, 'Filetype ' . $ext . ' not supported by Imagick.');
+
+			return '';
+		}
+
+		$tmp_file = tempnam(sys_get_temp_dir(), 'lychee') . '.jpeg';
+		Logs::notice(__METHOD__, __LINE__, 'Saving JPG of raw file to ' . $tmp_file);
+
+		$resWidth = $resHeight = 0;
+		$resWidth = $resHeight = 0;
+		$width = $photo->width;
+		$height = $photo->height;
+
+		try {
+			$this->imageHandler->scale($url, $tmp_file, $width, $height, $resWidth, $resHeight);
+		} catch (Exception $e) {
+			Logs::error(__METHOD__, __LINE__, 'Failed to create JPG from raw file ' . $url . $filename);
+
+			return '';
+		}
+
+		return $tmp_file;
+	}
+
+	/**
+	 * Creates smaller copies of Photo.
+	 *
+	 * @param Photo  $photo
+	 * @param string $type
+	 * @param int    $maxWidth
+	 * @param int    $maxHeight
+	 * @param string Path of the video frame
+	 *
+	 * @return bool
+	 */
+	public function resizePhoto(Photo $photo, string $type, int $maxWidth, int $maxHeight, string $frame_tmp = ''): bool
+	{
+		$width = $photo->width;
+		$height = $photo->height;
+
+		if ($frame_tmp === '') {
+			$filename = $photo->url;
+			$url = Storage::path('big/' . $filename);
+		} else {
+			$filename = $photo->thumbUrl;
+			$url = $frame_tmp;
+		}
+
+		// Both image sizes of the same type are stored in the same folder
+		// ie: medium and medium2x both belong in LYCHEE_UPLOADS_MEDIUM
+		$pathType = strtoupper($type);
+		if (($split = strpos($pathType, '2')) !== false) {
+			$pathType = substr($pathType, 0, $split);
+		}
+
+		$uploadFolder = Storage::path(strtolower($pathType) . '/');
+		if (Helpers::hasPermissions($uploadFolder) === false) {
+			Logs::notice(__METHOD__, __LINE__, 'Skipped creation of medium-photo, because ' . $uploadFolder . ' is missing or not readable and writable.');
```

Comment:
This is a generic code so it shouldn't mention `medium`. Either simply remove `medium` or replace with `$type`?

Reply:
fixed.